core/arch/arm/kernel/spmc_sp_handler.c

```diff
@@ -537,7 +537,10 @@ static void ffa_mem_retrieve(struct thread_smc_args *args,
 	receiver = sp_mem_get_receiver(caller_sp->endpoint_id, smem);
 
 	address_offset = READ_ONCE(retr_dsc->mem_access_array[0].region_offs);
-	assert(address_offset + sizeof(struct ffa_mem_region) <= tx_len);
+	if (address_offset + sizeof(struct ffa_mem_region) > tx_len) {
```

Comment:
We'd better check for overflow in case `address_offset` is a large number.

Reply:
Done